src/runtime/types.ts

```diff
@@ -208,91 +204,96 @@ export type ProviderLocal = {
+   * Configuration for the `refresh`-provider an extended version of the local provider.
+   * If set to undefined, local provider without refresh logic will be used.
    */
-  type: Extract<SupportedAuthProviders, 'refresh'>;
-  endpoints?: {
-    /**
-     * What method and path to call to perform the sign-in. This endpoint must return a token that can be used to authenticate subsequent requests.
-     *
-     * @default { path: '/refresh', method: 'post' }
-     */
-    refresh?: { path?: string; method?: RouterMethod };
-  };
-  /**
-   *  When refreshOnlyToken is set, only the token will be refreshed
-   *
-   * @default true
-   */
-  refreshOnlyToken?: boolean;
-
-  refreshToken?: {
-    /**
-     * How to extract the authentication-token from the sign-in response.
-     *
-     * E.g., setting this to `/refreshToken/bearer` and returning an object like `{ refreshToken: { bearer: 'THE_AUTH_TOKEN' }, timestamp: '2023' }` from the `signIn` endpoint will
-     * result in `nuxt-auth` extracting and storing `THE_AUTH_TOKEN`.
-     *
-     * This follows the JSON Pointer standard, see its RFC6901 here: https://www.rfc-editor.org/rfc/rfc6901
-     *
-     * @default '/refreshToken'  Access the `refreshToken` property of the sign-in response object
-     * @example /       Access the root of the sign-in response object, useful when your endpoint returns a plain, non-object string as the token
-     */
-    signInResponseRefreshTokenPointer?: string;
+  refresh?: {
     /**
-     * How to do a fetch for the refresh token.
-     *
-     * This is especially useful when you have an external backend signing tokens. Refer to this issue to get more information: https://github.com/sidebase/nuxt-auth/issues/635.
-     *
-     * ### Example
-     * Setting this to `/refresh/token` would make Nuxt Auth send the `POST /api/auth/refresh` with the following BODY: `{ "refresh": { "token": "..." } }
-     *
-     * ### Notes
-     * This follows the JSON Pointer standard, see its RFC6901 here: https://www.rfc-editor.org/rfc/rfc6901
+     * Whether the refresh logic of the local provider is active
      *
-     * @default '/refreshToken'
+     * @default false
      */
-    refreshRequestTokenPointer?: string;
+    isEnabled?: boolean,
     /**
-     * It refers to the name of the property when it is stored in a cookie.
+     * What method and path to call to perform the sign-in. This endpoint must return a token that can be used to authenticate subsequent requests.
      *
-     * @default auth.refresh-token
-     * @example auth._refresh-token
+     * @default { path: '/refresh', method: 'post' }
      */
-    cookieName?: string;
+    endpoint?: { path?: string; method?: RouterMethod };
     /**
-     * Maximum age to store the authentication token for. After the expiry time the token is automatically deleted on the application side, i.e., in the users' browser.
+     *  When refreshOnlyToken is set, only the token will be refreshed
```

Comment:
I think this sentence doesn't really match the code. Maybe you could rephrase it? Or maybe it should be renamed even?

I think we are refreshing the refresh token itself when this flag is set

Reply:
I checked [the code](https://github.com/sidebase/nuxt-auth/blob/821/feat/unify-local-refresh-providers/src/runtime/composables/local/useAuth.ts#L203) where we use this property:

```ts
  if (!config.refresh.refreshOnlyToken) {
    const refreshTokenPointer = config.refresh.token.signInResponseRefreshTokenPointer
    const extractedRefreshToken = jsonPointerGet(response, refreshTokenPointer)
    if (typeof extractedRefreshToken !== 'string') {
      console.error(
        `Auth: string token expected, received instead: ${JSON.stringify(extractedRefreshToken)}. ` +
         `Tried to find refresh token at ${refreshTokenPointer} in ${JSON.stringify(response)}`
      )
      return
    } else {
      rawRefreshToken.value = extractedRefreshToken
    }
  }
```

To me it looks like the following logic applies:

- `refreshOnlyToken` = `true`: We skip the step where we:
  - Extract the `refreshToken` using the `signInResponseRefreshTokenPointer` from the response of the `refresh.endpoint.path`
  - Set `rawRefreshToken` to the extracted refresh token
- `refreshOnlyToken` = `false`: We do the following:
  - Extract the `refreshToken` using the `signInResponseRefreshTokenPointer` from the response of the `refresh.endpoint.path`
  - Set `rawRefreshToken` to the extracted refresh token